ctapipe/io/magiceventsource.py

```diff
@@ -0,0 +1,1337 @@
+            True if the masked files are MAGIC data runs, False otherwise.
+
+        """
+
+        is_magic_root_file = True
+
+        file_list = glob.glob(file_mask)
+
+        for file_path in file_list:
+            try:
+                with uproot.open(file_path) as input_data:
+                    if 'Events' not in input_data:
+                        is_magic_root_file = False
+            except ValueError:
+                # uproot raises ValueError if the file is not a ROOT file
+                is_magic_root_file = False
+                pass
+
+        return is_magic_root_file
+
+    @staticmethod
+    def _get_run_number(file_name):
+        """
+        This internal method extracts the run number from
+        the specified file name.
+
+        Parameters
+        ----------
+        file_name: str
+            A file name to process.
+
+        Returns
+        -------
+        int:
+            A run number of the file.
+        """
+
+        mask = ".*\d+_M\d+_(\d+)\.\d+_.*"
+        parsed_info = re.findall(mask, file_name)
+
+        run_number = int(parsed_info[0])
+
+        return run_number
+
+    def _set_active_run(self, run_number):
+        """
+        This internal method sets the run that will be used for data loading.
+
+        Parameters
+        ----------
+        run_number: int
+            The run number to use.
+
+        Returns
+        -------
+
+        """
+
+        input_path = '/'.join(self.input_url.split('/')[:-1])
+        this_run_mask = input_path + '/*{:d}*root'.format(run_number)
+
+        run = dict()
+        run['number'] = run_number
+        run['read_events'] = 0
+        run['data'] = MarsDataRun(run_file_mask=this_run_mask)
+
+        return run
+
+    def _generator(self):
+        """
+        The default event generator. Return the stereo event
+        generator instance.
+
+        Returns
+        -------
+
+        """
+
+        return self._stereo_event_generator()
+
+    def _stereo_event_generator(self):
+        """
+        Stereo event generator. Yields DataContainer instances, filled
+        with the read event data.
+
+        Returns
+        -------
+
+        """
+
+        counter = 0
+
+        # Data container - is initialized once, and data is replaced within it after each yield
+        data = DataContainer()
+        data.meta['origin'] = "MAGIC"
+        data.meta['input_url'] = self.input_url
+        data.meta['is_simulation'] = False
+
+        # Telescopes with data:
+        tels_in_file = ["m1", "m2"]
+        tels_with_data = {1, 2}
+
+        # Loop over the available data runs
+        for run_number in self.run_numbers:
+
+            # Removing the previously read data run from memory
+            if self.current_run is not None:
+                if 'data' in self.current_run:
+                    del self.current_run['data']
+
+            # Setting the new active run
+            self.current_run = self._set_active_run(run_number)
```

Comment:
`self.current_run` seems to be only used locally in this method and not in other methods of the same class, so there seems to be no need for it to be an instance variable. I think it can just be a local.

Reply:
Done.